service/src/main/java/bio/terra/workspace/service/workspace/flight/aws/DeleteControlledAwsResourcesStep.java

```diff
@@ -0,0 +1,65 @@
+package bio.terra.workspace.service.workspace.flight.aws;
+
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.Step;
+import bio.terra.stairway.StepResult;
+import bio.terra.stairway.exception.RetryException;
+import bio.terra.workspace.db.ResourceDao;
+import bio.terra.workspace.service.iam.AuthenticatedUserRequest;
+import bio.terra.workspace.service.resource.controlled.ControlledResourceService;
+import bio.terra.workspace.service.resource.controlled.model.ControlledResource;
+import bio.terra.workspace.service.workspace.model.CloudPlatform;
+import java.util.List;
+import java.util.UUID;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * A step to delete all controlled AWS resources in a workspace. This reads the list of controlled
+ * AWS resources in a workspace from the WSM database.
+ */
+public class DeleteControlledAwsResourcesStep implements Step {
+
+  private final Logger logger = LoggerFactory.getLogger(DeleteControlledAwsResourcesStep.class);
+  private final ResourceDao resourceDao;
+  private final ControlledResourceService controlledResourceService;
+  private final UUID workspaceUuid;
+  private final AuthenticatedUserRequest userRequest;
+
+  public DeleteControlledAwsResourcesStep(
+      ResourceDao resourceDao,
+      ControlledResourceService controlledResourceService,
+      UUID workspaceUuid,
+      AuthenticatedUserRequest userRequest) {
+    this.resourceDao = resourceDao;
+    this.controlledResourceService = controlledResourceService;
+    this.workspaceUuid = workspaceUuid;
+    this.userRequest = userRequest;
+  }
+
+  @Override
+  public StepResult doStep(FlightContext flightContext)
+      throws InterruptedException, RetryException {
+
+    List<ControlledResource> controlledResourceList =
+        resourceDao.listControlledResources(workspaceUuid, CloudPlatform.AWS);
+    // TODO(TERRA-279): check permissions to delete
```

Comment:
I would expect a permission check on the delete workspace and the delete cloud context. I would not expect it to be done in this flight. By the time you are here, we should have checked permissions.